node-services/sharing-service/controllers/reportSharedWithMe.js

```diff
@@ -0,0 +1,36 @@
+const db = require('../dbInstance');
+const { validateEmail, validateUserParameter } = require('./validateParameters');
+
+const dbOperations = {
+  shareWithMe : async (reportObject) => {
+    if (!validateEmail(reportObject.email)) {
+      throw new Error("Invalid email");
+    }
+    if (!validateUserParameter(reportObject.id) || !validateUserParameter(reportObject.reportOwner)
+        || !validateUserParameter(reportObject.description)) {
+      throw new Error("Invalid parameter");
+    }
+    const collectionReference = db.collection('sharedWithMe');
+    const userReference = collectionReference.doc(reportObject.email);
+    const maxTries = 5;
+    tryRequest = async (currentAttempt, delay) => {
+      try {
+        await userReference.collection('reports').add({
+          reportId: reportObject.id,
+          reportDescription: reportObject.description,
+          reportOwner: reportObject.reportOwner
+        });
+      } catch (err) {
+        if (currentAttempt <= maxTries) {
+          setTimeout(async () => {
+            await tryRequest(currentAttempt + 1, delay * 2);
+          }, delay);
+        }
+        throw err;
```

Comment:
if the first try fails, it will go to catch, you will recursively call it one more time, (let's say it works then), control will get back to first call and this line will get executed and throw error (which it should not). Am i missing something?

Reply:
My bad, throwing error need to be in else part